typescript/s3-kms-cross-account-replication/stacks/step2-destination-account.ts

```diff
@@ -0,0 +1,137 @@
+import { Stack, StackProps, RemovalPolicy, Fn, CfnOutput, CfnParameter } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { Config } from '../config/configClass';
+import { Bucket, BucketEncryption, BlockPublicAccess, BucketAccessControl } from 'aws-cdk-lib/aws-s3'
+import { Key } from 'aws-cdk-lib/aws-kms';
+import { Effect, PolicyDocument, PolicyStatement, ArnPrincipal } from 'aws-cdk-lib/aws-iam';
+
+export class Step2DestinationAccount extends Stack {
+  public destinationS3Bucket: Bucket;
+  public destinationKmsKey: Key;
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const replicationRoleArn = new CfnParameter(this, 'replicationRoleArn', {
+      type: 'String',
+      description: 'The ARN of the replication role in the source account',
+    });
+
+    const destinationKmsKey = new Key(this, `s3-cross-account-replication-dest-key`, {
+      alias: Config.destinationKmsKeyAlias,
+      description: 'Key used for KMS Encryption for the destination s3 bucket for cross account replication',
+      policy: new PolicyDocument({
+        statements: [
+          new PolicyStatement({
+            sid: 'Enable IAM User Permissions',
+            effect: Effect.ALLOW,
+            principals: [new ArnPrincipal("arn:aws:iam::" + Config.destinationAccountId + ":root")],
+            actions: [
+              "kms:*",
+            ],
+            resources: ['*']
+          }),
+          new PolicyStatement({
+            sid: 'Enable Replication Permissions',
+            effect: Effect.ALLOW,
+            principals: [new ArnPrincipal(replicationRoleArn.valueAsString)],
+            actions: [
+              "kms:Encrypt",
+              "kms:Decrypt",
+              "kms:ReEncrypt*",
+              "kms:GenerateDataKey*",
+              "kms:DescribeKey",
+            ],
+            resources: ['*']
+          })
+        ]
+      })
+    });
+
+    // Create the destination S3 bucket
+    const destinationS3Bucket = new Bucket(this, 'destination-bucket-to-replicate-to', {
+      bucketName: Config.destinationBucketName,
+      removalPolicy: RemovalPolicy.DESTROY,
+      autoDeleteObjects: true,
+      versioned: true,
+      accessControl: BucketAccessControl.PRIVATE,
+      publicReadAccess: false,
+      blockPublicAccess: new BlockPublicAccess(BlockPublicAccess.BLOCK_ALL),
+      bucketKeyEnabled: true,
+      encryption: BucketEncryption.KMS,
+      encryptionKey: destinationKmsKey
+    });
+
+    // allow the principal to have all admin access to bucket
+    destinationS3Bucket.addToResourcePolicy(
+      new PolicyStatement({
+        sid: 'Set Admin Access',
+        effect: Effect.ALLOW,
+        principals: [new ArnPrincipal("arn:aws:iam::" + Config.destinationAccountId + ":root")],
```

Comment:
consider backticking this rather than using the overloaded addition: 
`arn:aws:iam::${Config.destinationAccountId}:root`

Reply:
Thanks, updated across file for consistency